pkg/status/combinedstatus-resolver.go

```diff
@@ -456,9 +478,26 @@ func (c *combinedStatusResolver) evaluateWorkStatusesPerBindingReadLocked(bindin
 				continue
 			}
 
-			// evaluate workstatus
 			csResolution := c.bindingNameToResolutions[bindingName][workStatus.sourceObjectIdentifier]
-			if csResolution.evaluateWorkStatus(c.celEvaluator, workStatus.wecName, workStatus.status) {
+			content := map[string]interface{}{}
+			content["status"] = workStatus.status
+
+			if csResolution.requiresSourceObjectMetaOrSpec() {
+				objMap, err := c.getObjectMetaAndSpec(workStatus.sourceObjectIdentifier)
+				if err != nil {
+					runtime2.HandleError(fmt.Errorf("failed to get meta & spec for source object %s: %w",
+						workStatus.sourceObjectIdentifier, err))
+				}
+
+				if objMap != nil {
+					for k, v := range objMap {
+						content[k] = v
+					}
+				}
+			}
+
+			// evaluate workstatus
+			if csResolution.evaluateWorkStatus(c.celEvaluator, workStatus.wecName, content) {
```

Comment:
There is enough duplicated code here now to be worth creating a procedural abstraction for.

Reply:
Fixed.

Reply:
Huh? I still see a copy of this code at new lines 314--335.

Reply:
Were you referring to abstracting the creation of the map only or also the call to evaluateWorkStatus?

Reply:
I used GitHub's feature where I can select a range of lines and put a comment on that range. That is the duplicated code I was referring to.

Reply:
In the current version of this PR, commit adcde7fd0d8bbce6d914570d33df6d9a41bc8495, much of the duplicated code has been swept into `c.getCombinedContentMap`. There remains some duplication outside that method: lines 314--320 and lines 460--466 are duplicates. Not a blocker, that is a small amount of duplicated material.